src/lib/containers/StackedRowHomebrew.js

```diff
@@ -269,19 +152,47 @@ export default class StackedRowHomebrew extends React.Component {
                             <div ref={measureRef}>
                                 {x_data.map(
                                     (obj, index) => {
+                                        let rgb = []
```

Comment:
should have a shared function for these calculations (get new RGB and deciding white/black text)